pkgs/by-name/ll/llama-cpp/package.nix

```diff
@@ -91,9 +93,10 @@ effectiveStdenv.mkDerivation (finalAttrs: {
 
   buildInputs = optionals effectiveStdenv.isDarwin darwinBuildInputs
     ++ optionals cudaSupport cudaBuildInputs
-    ++ optionals mpiSupport mpi
+    ++ optionals mpiSupport [ mpi ]
```

Comment:
This change looks good (and supports @drupol's claim that `optional` vs. `optionals` distinction leads to errors.)

I only use `llama-cpp` with `cuda` and so don't have the understanding to experiment with the BLAS ecosystem.